R/UK.R

```diff
@@ -496,6 +496,52 @@ UK <- R6::R6Class("UK",
           release_date = self$release_date
         )
       return(clean_data)
+    },
+
+    #' @description Specific tests for UK data. In addition to generic tests ran
+    #' by `DataClass$test()` data for NHS regions are downloaded and ran through
+    #' the same generic checks (test_cleaning, test_processing, test_return). If
+    #' download = TRUE or a snapshot file is not found, the nhs data is
+    #' downloaded and saved to the snapshot location provided. If an existing
+    #' snapshot file is found then this data is used in the next tests.
+    #' Tests data can be downloaded, cleaned, processed and returned.
+    #' @param self_copy R6class the object to test
+    #' @param download logical. To download the data (TRUE) or use a snapshot
+    #' (FALSE). Defaults to FALSE.
+    #' @param nhs_included_path character_array the path to save the downloaded
+    #' nhs data snapshot to.
+    specific_tests = function(self_copy, download = FALSE,
+                              nhs_included_path) {
+      if (missing(nhs_included_path)) {
+        message_verbose(
+          verbose = self_copy$verbose,
+          "nhs_included_path not provided, skipping test."
+        )
+        return(invisible(NULL))
+      }
+      if (self_copy$level == "1") {
+        self_copy$data_name <- "UK level 1 with 'nhsregions=TRUE'"
+        self_copy$nhsregions <- TRUE
+        source <- class(self_copy)[1]
+        if (!file.exists(nhs_included_path)) {
+          download <- TRUE
+        }
+        if (download) {
+          test_that(paste(data_name, " downloads sucessfully"), { # nolint
+            self_copy$download()
+            expect_s3_class(self_copy$data$raw$nhs, "data.frame")
+            expect_true(nrow(self_copy$data$raw$nhs) > 0)
+            expect_true(ncol(self_copy$data$raw$nhs) >= 2)
+          })
+          self_copy$data$raw$nhs <- slice_tail(self_copy$data$raw$nhs, n = 1000)
```

Comment:
we only want to test on a snapshot read from disk. If testing on the full data source should not take the tail (I think)?

Reply:
So add another parameter, such as `full` which if TRUE saves the full dataset to work on and if FALSE saves just the snapshots for the tests?

Reply:
I was actually just thinking of some logic so that if working from the snapshot then it is partial data and if working from the raw download it is complete but yes another arg could be good but not required.

Reply:
Sorry, I'm confused I thought the tests always work on a snapshot of the raw data?